# - Change available size from 32, 40, 48, 64 to 24, 30, 36, 48
# - Remove command avalibility checks because `which` could not detect `nativeBuildInputs`
# - Silence inkscape
--- a/build.sh
+++ b/build.sh
@@ -1,48 +1,13 @@
 #! /usr/bin/env bash
 
-# check command avalibility
-has_command() {
-  "$1" -v $1 > /dev/null 2>&1
-}
-
-if [ ! "$(which xcursorgen 2> /dev/null)" ]; then
-  echo xorg-xcursorgen needs to be installed to generate the cursors.
-  if has_command zypper; then
-    sudo zypper in xorg-xcursorgen
-  elif has_command apt; then
-    sudo apt install xorg-xcursorgen
-  elif has_command dnf; then
-    sudo dnf install -y xorg-xcursorgen
-  elif has_command dnf; then
-    sudo dnf install xorg-xcursorgen
-  elif has_command pacman; then
-    sudo pacman -S --noconfirm xorg-xcursorgen
-  fi
-fi
-
-if [ ! "$(which inkscape 2> /dev/null)" ]; then
-  echo inkscape needs to be installed to generate the cursors.
-  if has_command zypper; then
-    sudo zypper in inkscape
-  elif has_command apt; then
-    sudo apt install inkscape
-  elif has_command dnf; then
-    sudo dnf install -y inkscape
-  elif has_command dnf; then
-    sudo dnf install inkscape
-  elif has_command pacman; then
-    sudo pacman -S --noconfirm inkscape
-  fi
-fi
-
 function create {
 	cd "$SRC"
 	mkdir -p x1 x1_25 x1_5 x2
 	cd "$SRC"/$1
-	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x1/${0%.svg}.png" -w 32 -h 32 $0' {} \;
-	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x1_25/${0%.svg}.png" -w 40 -w 40 $0' {} \;
-	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x1_5/${0%.svg}.png" -w 48 -w 48 $0' {} \;
-	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x2/${0%.svg}.png" -w 64 -w 64 $0' {} \;
+	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x1/${0%.svg}.png" -w 24 -h 24 $0 &>/dev/null' {} \;
+	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x1_25/${0%.svg}.png" -w 30 -w 30 $0 &>/dev/null' {} \;
+	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x1_5/${0%.svg}.png" -w 36 -w 36 $0 &>/dev/null' {} \;
+	find . -name "*.svg" -type f -exec sh -c 'inkscape -o "../x2/${0%.svg}.png" -w 48 -w 48 $0 &>/dev/null' {} \;
 
 	cd $SRC
 
